traffic_ops/experimental/auth/auth.go

```diff
@@ -132,83 +150,197 @@ func InitializeDatabase(username, password, dbname, server string, port uint) (*
+				return
+			}
+
+	    	hasher := sha1.New()
+	    	hasher.Write([]byte(login.Password))
+	    	hashedPassword := fmt.Sprintf("%x", hasher.Sum(nil))
+
+			if len(tmUserList) == 0 || tmUserList[0].Password != string(hashedPassword) {
+				Logger.Printf("Invalid username/password, username %s", login.Username)
+				http.Error(w, "Invalid username/password", http.StatusUnauthorized)
+				return
+			}
+
+			// We have validated the user's password, now lets get the user's roles
+			stmt, err = db.PrepareNamed("SELECT role_id FROM user_role WHERE user_id=:id")
+			if err != nil {
+				Logger.Println("Database error: ", err.Error())
+				http.Error(w, "Database error: "+err.Error(), http.StatusInternalServerError)
+				return
+			}
+
+			err = stmt.Select(&userRoleList, tmUserList[0])
+			if err != nil {
+				Logger.Println("Database error: ", err.Error())
+				http.Error(w, "Database error: "+err.Error(), http.StatusInternalServerError)
+				return
+			}
+
+			rolesIds := []int{}
+			for _, elem := range userRoleList {
+				rolesIds = append(rolesIds, elem.RoleId)
+			}
+
+			// Get user's capabilities according to the user's roles
+			sql, args, err := sqlx.In("SELECT cap_name FROM role_capability WHERE role_id IN (?)", rolesIds)
+			if err != nil {
+				Logger.Println("Database error: ", err.Error())
+				http.Error(w, "Database error: "+err.Error(), http.StatusInternalServerError)
+				return
+			}
+
+			// Replace the "?" bindvar syntax with DB specific syntax ($1, $2, ... for PostgreSQL)
+			sql = db.Rebind(sql)
+
+			stmt1, err := db.Preparex(sql)
+			if err != nil {
+				Logger.Println("Database error: ", err.Error())
+				http.Error(w, "Database error: "+err.Error(), http.StatusInternalServerError)
+				return
+			}
+
+			err = stmt1.Select(&capList, args...)
+			if err != nil {
+				Logger.Println("Database error: ", err.Error())
+				http.Error(w, "Database error: "+err.Error(), http.StatusInternalServerError)
+				return
+			}
+
+			capabilities := []string{}
+			for _, elem := range capList {
+				capabilities = append(capabilities, elem.CapName)
+			}
+
+			/////////////////////////////////////////////////////////////////////////////////////////////////
+			// LEGACY: Perform login against legacy TO. This is required until AAA is disabled in TO
+			legacyResp, err := LegacyTOLogin(login, config.LegacyLoginURL, w);
+			if err != nil {
+				msg := fmt.Sprintf("Legacy TO login error: %s", err.Error())
+				Logger.Println(msg)
+				http.Error(w, msg, http.StatusInternalServerError)
+				return;
+			}
+
+			if legacyResp.StatusCode != 200 {
+				msg := fmt.Sprintf("Legacy TO login returned bad status: %s", legacyResp.Status)
+				Logger.Println(msg)
+				http.Error(w, msg, http.StatusInternalServerError)
+				return
+			}
+
+			legacyCookies := legacyResp.Cookies()
+
+			if (legacyCookies == nil) || (len(legacyCookies) != 1) || (legacyCookies[0].Name != "mojolicious") {
```

Comment:
This piece of code is a patch for a temporary situation where 2 AAA models live together simultaneously. IMO it should look like a patch... simple, problem specific, self contained and verbose. Putting "mojolicious" in a const goes against that.